hawkbit-security-core/src/main/java/org/eclipse/hawkbit/security/SystemSecurityContext.java

```diff
@@ -117,6 +117,28 @@ public SystemSecurityContext(final TenantAware tenantAware) {
         }
     }
 
+    public <T> T runAsSystemAsTenantAsUser(final Callable<T> callable, final String tenant, final String username) {
```

Comment:
Not sure if this is intended or the name is just a bit misleading: This method will give the provided username "superpower" by means all permissions! It is not running with the permissions assigned to the user itself

Reply:
you're right, maybe a better approach is to just `runAsTenantAsUser` and read the permissions for that user from somewhere. The system scope is not needed I think